src/Orleans/Streams/Core/StreamSubscriptionManager.cs

```diff
@@ -0,0 +1,55 @@
+﻿using Orleans.Runtime;
+using System;
+using System.Collections.Concurrent;
+using System.Collections.Generic;
+using System.Threading.Tasks;
+
+namespace Orleans.Streams.Core
+{
+    internal class StreamSubscriptionManager: IStreamSubscriptionManager
+    {
+        private IStreamPubSub streamPubSub;
+        private StreamPubSubType type;
+        private string streamProviderName;
+        //potential memory optimization on this
+        private ConcurrentDictionary<Guid, StreamSubscription> allSubscriptions;
+        public StreamSubscriptionManager(IStreamPubSub pubsub, StreamPubSubType t, string streamProviderName)
```

Comment:
I would prefer the StreamSubscriptionManager not have special case logic for each pubsub type.  Let's discuss how to avoid this.